exercises/run-length-encoding/RunLengthEncodingTest.cs

```diff
@@ -1,60 +1,84 @@
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Encode_multiple_whitespace_mixed_in_string()
+    {
+        Assert.Equal("2 hs2q q2w2 ", RunLengthEncoding.Encode("  hsqq qww  "));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Encode_lowercase_characters()
+    {
+        Assert.Equal("2a3b4c", RunLengthEncoding.Encode("aabbbcccc"));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Decode_empty_string()
     {
-        const string input = "AABBBCCCC";
-        const string expected = "2A3B4C";
-        Assert.Equal(expected, RunLengthEncoding.Encode(input));
+        Assert.Equal("", RunLengthEncoding.Decode(""));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Decode_simple()
+    public void Decode_single_characters_only()
     {
-        const string input = "2A3B4C";
-        const string expected = "AABBBCCCC";
-        Assert.Equal(expected, RunLengthEncoding.Decode(input));
+        Assert.Equal("XYZ", RunLengthEncoding.Decode("XYZ"));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Encode_with_single_values()
+    public void Decode_string_with_no_single_characters()
     {
-        const string input = "WWWWWWWWWWWWBWWWWWWWWWWWWBBBWWWWWWWWWWWWWWWWWWWWWWWWB";
-        const string expected = "12WB12W3B24WB";
-        Assert.Equal(expected, RunLengthEncoding.Encode(input));
+        Assert.Equal("AABBBCCCC", RunLengthEncoding.Decode("2A3B4C"));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Decode_with_single_values()
+    public void Decode_single_characters_with_repeated_characters()
     {
-        const string input = "12WB12W3B24WB";
-        const string expected = "WWWWWWWWWWWWBWWWWWWWWWWWWBBBWWWWWWWWWWWWWWWWWWWWWWWWB";
-        Assert.Equal(expected, RunLengthEncoding.Decode(input));
+        Assert.Equal("WWWWWWWWWWWWBWWWWWWWWWWWWBBBWWWWWWWWWWWWWWWWWWWWWWWWB", RunLengthEncoding.Decode("12WB12W3B24WB"));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Encode_and_then_decode()
+    public void Decode_multiple_whitespace_mixed_in_string()
     {
-        const string input = "zzz ZZ  zZ";
-        const string expected = "zzz ZZ  zZ";
-        Assert.Equal(expected, RunLengthEncoding.Decode(RunLengthEncoding.Encode(input)));
+        Assert.Equal("  hsqq qww  ", RunLengthEncoding.Decode("2 hs2q q2w2 "));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Encode_unicode()
+    public void Decode_lower_case_string()
     {
-        const string input = "⏰⚽⚽⚽⭐⭐⏰";
-        const string expected = "⏰3⚽2⭐⏰";
-        Assert.Equal(expected, RunLengthEncoding.Encode(input));
+        Assert.Equal("aabbbcccc", RunLengthEncoding.Decode("2a3b4c"));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Decode_unicode()
+    public void Consistency_encode_followed_by_decode_gives_original_string()
```

Comment:
@ErikSchierboom this test case is to verify that `x == Decode(Encode(x))`., but the way that I have it now obviously isn't working. Any thoughts on how that should be handled in the generator?